src/solidity-frontend/solidity_convert.cpp

```diff
@@ -0,0 +1,1831 @@
+  }
+
+  return adjusted_expr;
+}
+
+nlohmann::json solidity_convertert::make_array_elementary_type(const nlohmann::json& type_descrpt)
+{
+  // Function used to extract the elementary type of array
+  // In order to keep the consistency and maximum the reuse of get_type_description function,
+  // we used ["typeDescriptions"] instead of ["typeName"], despite the fact that the latter contains more information.
+  // Although ["typeDescriptions"] also contains all the information needed, we have to do some
+  // pre-processing, e.g. extract the
+  nlohmann::json elementary_type;
+  if (type_descrpt["typeString"].get<std::string>().find("uint8") != std::string::npos)
+  {
+    auto j = R"(
+      {
+        "typeIdentifier": "t_uint8",
+        "typeString": "uint8"
+      }
+    )"_json;
+    elementary_type = j;
+  }
+  else
+  {
+    assert(!"Unsupported array elementary type");
+  }
+
+  return elementary_type;
+}
+
+nlohmann::json solidity_convertert::make_array_to_pointer_type(const nlohmann::json& type_descrpt)
+{
+  // Function to replace the content of ["typeIdentifier"] with "ArrayToPtr"
+  // All the information in ["typeIdentifier"] should also be available in ["typeString"]
+  std::string type_identifier = "ArrayToPtr";
+  std::string type_string = type_descrpt["typeString"].get<std::string>();
+
+  std::map<std::string, std::string> m = {
+    {"typeIdentifier", type_identifier},
+    {"typeString", type_string}
+  };
+  nlohmann::json adjusted_type = m;
+
+  return adjusted_type;
+}
+
+std::string solidity_convertert::get_array_size(const nlohmann::json& type_descrpt)
+{
+  const std::string s = type_descrpt["typeString"].get<std::string>();
+  std::regex rgx(".*\\[([0-9]+)\\]");
+  std::string the_size;
+
+  std::smatch match;
+  if (std::regex_search(s.begin(), s.end(), match, rgx))
+  {
+    std::ssub_match sub_match = match[1];
+    the_size = sub_match.str();
+  }
+  else
+  {
+    assert(!"Unsupported - Missing array size in type descriptor. Detected dynamic array?");
+  }
+
+  return the_size;
+}
+
+// debug functions
+void solidity_convertert::print_json(const nlohmann::json &json_in)
+{
+  printf("### JSON: ###\n");
+  std::cout << std::setw(2) << json_in << '\n'; // '2' means 2x indentations in front of each line
+  printf("\n");
+}
+
+void solidity_convertert::print_json_element(const nlohmann::json &json_in, const unsigned index,
+    const std::string &key, const std::string& json_name)
+{
+  printf("### %s element[%u] content: key=%s, size=%lu ###\n",
+      json_name.c_str(), index, key.c_str(), json_in.size());
+  std::cout << std::setw(2) << json_in << '\n'; // '2' means 2x indentations in front of each line
+  printf("\n");
+}
+
+void solidity_convertert::print_json_array_element(const nlohmann::json &json_in,
+    const std::string& node_type, const unsigned index)
+{
+  printf("### node[%u]: nodeType=%s ###\n", index, node_type.c_str());
```

Comment:
Why don't we use our messaging interface based on the fmt library?

Reply:
Same reason as explained in the comment of Line 1828 (function `print_json_stmt_element`)